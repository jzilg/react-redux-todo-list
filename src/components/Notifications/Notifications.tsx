import React, { ReactElement } from 'react'
import Notification from '../../entities/notification.interface'
import style from './notifications.scss'

interface Props {
    notifications: Notification[]
    removeNotification: Function
}

const Notifications = ({ notifications, removeNotification }: Props): ReactElement<{}> => {
    if (!notifications.length) {
        return null
    }

    const notificationMessageElements = notifications.map((notification) => {
        const { id, type, message } = notification

        function clickHandler(): void {
            removeNotification(id, 'button on notification itself clicked')
        }

        return (
            <li key={id} className={style[type]}>
                <div className={style.container}>
                    <span>
                        {message}
                    </span>
                    <button onClick={clickHandler} className={style.button} type="button">
                        âœ•
                    </button>
                </div>
            </li>
        )
    })

    return (
        <ul className={style.list}>
            {notificationMessageElements}
        </ul>
    )
}

export default Notifications
