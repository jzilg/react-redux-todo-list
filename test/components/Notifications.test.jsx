import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { shallowToJson } from 'enzyme-to-json'
import Notifications from '../../src/components/notifications'

Enzyme.configure({ adapter: new Adapter() })

describe('Icon', () => {
    const notifications = [
        {
            id: 0,
            type: 'error',
            message: 'Error',
        },
    ]

    it('should return null if notifications array is empty', () => {
        const emptyNotifications = []
        const component = shallow(<Notifications notifications={emptyNotifications} />)
        expect(shallowToJson(component)).toMatchSnapshot()
    })

    it('should render correctly', () => {
        const component = shallow(<Notifications notifications={notifications} />)
        expect(shallowToJson(component)).toMatchSnapshot()
    })

    it('should call removeNotification if button is clicked', () => {
        const removeNotification = jest.fn()
        const component = shallow(
            <Notifications
                notifications={notifications}
                removeNotification={removeNotification}
            />
        )
        component.find('button').simulate('click')
        expect(removeNotification.mock.calls.length).toBe(1)
    })
})
