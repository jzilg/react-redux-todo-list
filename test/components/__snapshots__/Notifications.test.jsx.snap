// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render correctly 1`] = `
<ul>
  <li
    key="0"
  >
    <div>
      <span>
        Error
      </span>
      <button
        onClick={[Function]}
        type="button"
      >
        âœ•
      </button>
    </div>
  </li>
</ul>
`;

exports[`Icon should return null if notifications array is empty 1`] = `""`;
